libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/files/FileRequests.java

```diff
@@ -234,21 +217,42 @@ public static RestRequest addFileShare(String fileId, String entityId, String sh
      */
     public static RestRequest deleteFileShare(String shareId) {
         validateSfdcId(shareId);
-        return new RestRequest(RestMethod.DELETE, CONTENT_DOCUMENT_LINK + "/" + shareId, null);
+        return new RestRequest(RestMethod.DELETE, getContentDocumentLinkPath() + "/" + shareId, null);
     }
 
-    private static HttpEntity makeFileShare(String fileId, String entityId, String shareType) {
+    /**
+     * Build a request that can upload a new file to the server, this will
+     * create a new file at version 1.
+     *
+     * @param theFile
+     *            The path of the local file to upload to the server.
+     * @param name
+     *            The name of this file.
+     * @param title
+     *            The title of this file.
+     * @param description
+     *            A description of the file.
+     * @param mimeType
+     *            The mime-type of the file, if known.
+     * @return A RestRequest that can perform this upload.
+     *
+     * @throws UnsupportedEncodingException
+     */
+    public static RestRequest uploadFile(File theFile, String name, String title, String description, String mimeType) throws UnsupportedEncodingException {
```

Comment:
uploadFile is back, and it returns a RestRequest, so that it can be sent synchronously or asynchronously like any other RestRequest.